### PR TITLE
chore(cli): limit `pipeline delete` to prompt about deleting secret only when secret exists

### LINKED ISSUE #1827: pipeline: skip secret prompt when deleting CC pipeline
The secret field is empty for CodeCommit-sourced pipelines. 
When deleting these pipelines, the user still gets prompted, "Are you sure you want to delete the source secret  associated with pipeline [pipeline name]? This should happen only for GitHub tokens and other pipeline secrets.

### PR DESCRIPTION
**Pull request by huanjani**
Now that there are non-GitHub source options, there will be pipeline sources without secrets. `pipeline delete` should prompt "Are you sure you want to delete the source secret...?" only when there is a secret. 

Fixes https://github.com/aws/copilot-cli/issues/1827.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/pipeline_delete.go [hunk 1]
```diff
@@ -118,8 +118,10 @@ func (o *deletePipelineOpts) Ask() error {
 
 // Execute deletes the secret and pipeline stack.
 func (o *deletePipelineOpts) Execute() error {
-	if err := o.deleteSecret(); err != nil {
-		return err
+	if o.PipelineSecret != "" {
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_delete.go]: Do we not have tests for pipeline delete 😨

Comment (huanjani) [reply internal/pkg/cli/pipeline_delete.go]: Adding momentarily!

Comment (bvtujo) [reply internal/pkg/cli/pipeline_delete.go]: Can we make this idempotent by wrapping a SecretNotFound error from the underlying client call in `deleteSecret` on L173?

Comment (huanjani) [reply internal/pkg/cli/pipeline_delete.go]: [discussed offline. moved conditional from `Execute()` into `deleteSecret()`]

File: internal/pkg/cli/pipeline_delete_test.go [hunk 2]
```diff
@@ -191,6 +191,19 @@ func TestDeletePipelineOpts_Execute(t *testing.T) {
 
 		wantedError error
 	}{
+		"skips delete secret confirmation (and deletion attempt) if there is no secret": {
+			inAppName:      testAppName,
+			inPipelineName: testPipelineName,
+			setupMocks: func(mocks deletePipelineMocks) {
+				gomock.InOrder(
+					mocks.prog.EXPECT().Start(fmt.Sprintf(fmtDeletePipelineStart, testPipelineName, testAppName)),
+					mocks.deployer.EXPECT().DeletePipeline(testPipelineName).Return(nil),
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_delete_test.go:201]: a cool thing that we can do with mocks is also checking if a method was never invoked, for example:
`mocks.secretsmanager.DeleteSecret(gomock.Any()).Times(0)`

**Review by efekarakus — APPROVED** [review]

**Review by bvtujo — APPROVED** [review]

**Pull request merged**